vscode/src/utilities/configuration.ts

```diff
@@ -201,3 +204,91 @@ export async function updateGetSolutionMaxIterations(value: number): Promise<voi
     vscode.ConfigurationTarget.Workspace,
   );
 }
+
+interface GenAIConfigStatus {
+  configured: boolean;
+  keyMissing: boolean;
+  usingDefault: boolean;
+  activeKey?: string;
+}
+
+export function getGenAIConfigStatus(filepath: string): GenAIConfigStatus {
+  try {
+    const fileContents = fs.readFileSync(filepath, "utf8");
+    const config = yaml.load(fileContents) as any;
+
+    const models = config?.models ?? {};
+    const activeConfig = config?.active;
+
+    if (!activeConfig || typeof activeConfig !== "object") {
+      return { configured: false, keyMissing: false, usingDefault: true };
+    }
+
+    // Manually find the model key whose value matches the active config
+    let resolvedActiveKey: string | undefined = undefined;
+    for (const [key, model] of Object.entries(models)) {
+      if (deepEqual(model, activeConfig)) {
+        resolvedActiveKey = key;
+        break;
+      }
+    }
+
+    const env = activeConfig.environment ?? {};
+    const apiKey = env.OPENAI_API_KEY?.trim?.();
+
+    return {
+      configured: Boolean(apiKey),
+      keyMissing: apiKey === "" || apiKey === undefined,
+      usingDefault:
+        resolvedActiveKey === "OpenAI" &&
+        activeConfig?.args?.model === "gpt-4o" &&
+        Object.values(env).every((v: any) => !v || v.trim() === ""),
+      activeKey: resolvedActiveKey,
+    };
+  } catch (err) {
+    console.error("Error parsing GenAI config:", err);
+    return { configured: false, keyMissing: false, usingDefault: true };
+  }
+}
+
+export function updateAnalysisConfig(draft: ExtensionData, settingsPath: string): void {
+  const config = vscode.workspace.getConfiguration("konveyor.analysis");
+  const labelSelector = config.get<string>("labelSelector");
+  const UNCONFIGURED_VALUES = [undefined, "discovery", "(discovery)"];
```

Comment:
Just use the `getConfigLabelSelector` that already exists? Could also use `getConfigCustomRules` for that bit.